datastore/README.rst

```diff
@@ -120,10 +120,46 @@ The full list of classes which may be overridden in this way is:
         key_kind.path_element_kind = PathElement
         query_result_batch_kind = QueryResultBatch
         query_result_batch_kind.entity_result_kind = EntityResult
+        value_kind = Value
+        value_kind.key_kind = Key
 
 You can then drop-in the ``MyVeryCustomDatastore`` class anywhere where you
 previously used ``Datastore``.
 
+If you want to use a custom Key kind and be able to use queries with it, you
+will need to implement your own ``Value``, ``Query`` and ``GQLQuery`` classes
+and wire them up to the rest of the custom classes:
+
+.. code-block:: python
+
+    class MyKey:
+        pass
+
+    class MyValue(gcloud.aio.datastore.Value):
+        key_kind = MyKey
+
+    class MyEntity(gcloud.aio.datastore.Entity):
+        key_kind = MyKey
+        value_kind = MyValue
+
+    class MyEntityResult(gcloud.aio.datastore.EntityResult):
+        entity_kind = MyEntity
+
+    class MyQueryResultBatch(gcloud.aio.datastore.QueryResultBatch):
+        entity_result_kind = MyEntityResult
+
+    class MyDatastore(gcloud.aio.datastore.Datastore):
+        key_kind = MyKey
```

Comment:
This is starting to look a bit messy and that could probably be cleaned up.

For example, in `MyEntity`, it shouldn't be possible to change `value_kind` without changing `key_kind` because things will break. So we should be able to remove `Entity.key_kind` completely and rely on `value_kind.key_kind` inside that class.
It's probably true for some combinations of `Datastore`'s attributes as well. It would require some careful thinking before removing some of them but I'm sure it's possible.

But that's not backward-compatible.